.github/workflows/spack.yml

```
@@ -81,25 +63,99 @@ jobs:
+              unify: true
+              targets:
+                granularity: generic
+            packages:
+              petsc:
+                require: ~hdf5
+              rocblas:
+                require: ~tensile
+            repos:
+            - spack/local
+            mirrors:
+              spack:
+                binary: true
+                url: https://binaries.spack.io/${SPACK_VERSION}
+              local-buildcache:
+                binary: true
+                autopush: true
+                url: oci://${{ env.REGISTRY }}-${SPACK_VERSION}
+                signed: false
+                access_pair:
+                  id_variable: GITHUB_USER
+                  secret_variable: GITHUB_TOKEN
+          EOF
+
+      - name: Configure External Packages
         run: |
-          # Configure environment
-          export NUM_PROC_TEST=$(nproc 2> /dev/null || sysctl -n hw.ncpu)
-          if [[ "$NUM_PROC_TEST" -gt "$NUM_PROC_TEST_MAX" ]]; then
-            NUM_PROC_TEST=$NUM_PROC_TEST_MAX
-          fi
-          . /opt/spack/share/spack/setup-env.sh
-          spack load palace
+          # These cause build issues if built as externals
+          #   - python : often distributed python isn't feature complete / not all dependencies get detected
+          #   - OpenSSL / OpenSSH : since they are in /usr, spack struggles. It's common to rebuild these
+          #   - ncurses / bzip2 / xz / curl : caused build issues. We pull these from GHCR cache after first build
+          spack -e . external find --all \
+            --exclude openssl \
+            --exclude openssh \
+            --exclude python \
+            --exclude ncurses \
+            --exclude bzip2 \
+            --exclude xz \
+            --exclude curl
+
+      - name: Configure Compilers
+        run: spack -e . compiler find && spack -e . compiler list
+
+      - name: Configure Binary Mirror Keys
+        run: |
+          # If we cached these, that would be faster and safer
+          spack -e . buildcache keys --install --trust
 
+      - name: Bootstrap
+        run: spack -e . bootstrap now
+
+      - name: Concretize
+        # In theory we can re-use a concretization and pin a spack to speed this up.
+        # Unfortunately it then becomes difficult to know when to re-concretize.
+        run: |
+          # Using `spack develop` in order to have an in-source build
+          spack -e . develop --path=$(pwd) local.palace@git."${{ github.head_ref || github.ref_name }}"=develop
+          spack -e . concretize -f
+
+      - name: Build Dependencies
+        run: |
+          spack -e . install --only-concrete --no-check-signature --fail-fast --show-log-on-error --only dependencies
+
+      - name: Build Palace
+        # Build palace from source using this current directory
+        # We use `--no-cache` in order to force a rebuild
+        run: spack -e . install --only-concrete --keep-stage --show-log-on-error --only package --no-cache
```

```suggestion
        run: spack -e . install --only-concrete --keep-stage --verbose --only package --no-cache
```

I just tried to compare build logs for #360 against the passing builds in this PR, and realized that this change might be nice.

@hughcars up to you, but these are just preference at this point.